### PR TITLE
Logger init panic

### LINKED ISSUE #56: 日志库 Panic
通过 InitGlobalRoller 更新 roller 从 log.Roller{MaxTime: 24 * 60 * 60} 到 log.Roller{MaxSize: 100, MaxAge: 7, MaxBackups: 10} 发生 panic。
目前只支持：修改 MaxSize 字段吗？而且 MaxSize 这个字段不可以修改为0。

### PR DESCRIPTION
**Pull request by YIDWang**
修复日志的 panic 
Fixes #56

### CODE REVIEW

File: log/roller.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ var (
 	lumberjacksLocker sync.Mutex
 
 	errInvalidRollerParameter = errors.New("invalid roller parameter")
+	errTimeZero               = errors.New("time is not zero")
```
Comment (doujiang24) [inline log/roller.go:44]:
```suggestion
	errTimeZero               = errors.New("time should not be zero")
```

Comment (YIDWang) [reply log/roller.go]: fix

File: log/roller_test.go [hunk 2]
```diff
@@ -76,14 +77,16 @@ func TestParseRoller(t *testing.T) {
...
 
-	praseArgs = "size=100"
-	err = InitGlobalRoller(praseArgs)
-	if err != nil {
-		t.Errorf("ParseRoller failed")
-	}
-	if defaultRoller.MaxSize != 100 || defaultRoller.Compress != false {
-		t.Errorf("ParseRoller failed")
-	}
+	/*
```
Comment (doujiang24) [inline log/roller_test.go:80]: we'd better fix it instead of removing it. It's not a good idea to remove the existing test case.

Comment (YIDWang) [reply log/roller_test.go]: fix

**Comment by doujiang24** [comment]
@YIDWang Could you please paste the backtrace of panic? I'm not sure this PR is the proper way to fix it.

**Comment by YIDWang** [comment]
image
```
func TestGetOrCreate(t *testing.T) {
	roller := &Roller{
		MaxAge:     1,
		MaxSize:    10,
		MaxBackups: 3,
	}
	log, err := GetOrCreateLogger("/tmp/rollertest/test.log", roller)
	assert.NoError(t, err)

	log2, err := GetOrCreateLogger("/tmp/rollertest/test_bak.log", nil)
	assert.NoError(t, err)

	output := bytes.NewBuffer(nil)
	for output.Len() < 8*1024*1024 {
		output.WriteString(time.Now().String())
	}
	err = InitGlobalRoller("size=100 age=7 keep=10") //  Maxtime=10 change "size=100 age=7 keep=10"  panic
	assert.Error(t, err)

	for i := 0; i < 100; i++ {
		log.Write(output.Bytes())
		log2.Write(output.Bytes())
	}
	log.Close()
}
```

**Comment by doujiang24** [comment]
@YIDWang Thanks, got it.
I think the proper way may be to reopen the logger.
otherwise, we have no way to change the `MaxTime` to zero which may break the existing behavior, it's allowed and useful before the logger started.

**Pull request closed without merging**